### PR TITLE
Fix GN build to compile gen_snapshot on Windows.

### PR DESCRIPTION
**Pull request by goderbauer**
This is essentially a backport of https://github.com/dart-lang/sdk/commit/9a6797e6ba51846b51dce92e19658af3414db50e, where the Dart SDK is doing the same based on a copy of Flutter's build system.

Required for https://github.com/flutter/flutter/issues/138

### CODE REVIEW

File: build/config/compiler/BUILD.gn [hunk 1]
```diff
@@ -436,8 +388,10 @@ config("compiler") {
   # Tracked in https://fuchsia.atlassian.net/browse/TO-61
   if (is_linux && current_cpu == "x86") {
     cc_std = [ "-std=c++11" ]
-  } else {
+  } else if (!is_win) {
     cc_std = [ "-std=c++14" ]
+  } else {
+    cc_std = []
```
Comment (abarth) [inline build/config/compiler/BUILD.gn]: You want to put something there, even if it is `-std=c++11`

Comment (abarth) [inline build/config/compiler/BUILD.gn]: (Or whatever the VS equivalent is)

Comment (goderbauer) [inline build/config/compiler/BUILD.gn]: Done.

**Comment by abarth** [comment]
LGTM

**Pull request merged**